### PR TITLE
[DependencyInjection] Add support for `#[Autowire(lazy: true)]`

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

This is a feature that I anticipated during my talk at SymfonyCon DisneyLand.

This PR adds support for using `#[Autowire(lazy: true)]` on an argument to tell the container to generate a lazy service and pass it to that argument.

This works by creating a lazy service from the autowiring alias: if the argument targets `FooInterface`, a lazy service is declared (named `.lazy.FooInterface`) and injected. The original `FooInterface` remains unchanged, aka non-lazy.

This allows services that don't always consume their deps to initialize them only when actually needed.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php [hunk 1]
```diff
@@ -1719,6 +1720,29 @@ public function testAutowireClosure()
         $this->assertInstanceOf(Foo::class, $fooClone = ($bar->buz)());
         $this->assertNotSame($container->get('foo'), $fooClone);
     }
+
+    public function testLazyAutowireAttribute()
+    {
+        $container = new ContainerBuilder();
+        $container->register('foo', Foo::class)
+        ->setPublic('true');
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php]: indent :)

File: src/Symfony/Component/DependencyInjection/Attribute/Autowire.php [hunk 2]
```diff
@@ -41,8 +41,12 @@ public function __construct(
         string $expression = null,
         string $env = null,
         string $param = null,
+        public bool $lazy = false,
     ) {
-        if (!(null !== $value xor null !== $service xor null !== $expression xor null !== $env xor null !== $param)) {
+        if ($lazy && null !== ($expression ?? $env ?? $param)) {
+            throw new LogicException('#[Autowire] attribute cannot be $lazy and use $env, $param nor $value.');
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Attribute/Autowire.php]:
```suggestion
            throw new LogicException('#[Autowire] attribute cannot be $lazy and use $env, $param, or $value.');
```

**Review by fabpot — APPROVED** [review]
Minor comments

**Comment by fabpot** [comment]
Needs a rebase.

**Comment by nicolas-grekas** [comment]
Rebased + comments addressed, thanks.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**